### ISSUE TITLE
k3sup fails to complete installation in LXC

### ISSUE THREAD

**Issue by cmonty14** [alexellis/k3sup#326]
## Expected Behaviour

Installation in LXC completes w/o error.

## Current Behaviour

Installation in LXC fails when running command `k3sup install --ip $SERVER_IP --user $USER --cluster --k3s-version v1.21.0+k3s1`

## Steps to Reproduce (for bugs)

1. Execute k3setup install

## Context

This is the first attempt of "[Create a multi-master (HA) setup with embedded etcd](https://github.com/alexellis/k3sup#create-a-multi-master-ha-setup-with-embedded-etcd)"

This is the output of journalctl -xe executed in LXC:
```
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.285092000+02:00" level=info msg="Running kube-scheduler --address=127.0.0.1 --bind-address=127>
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.285626765+02:00" level=info msg="Running kube-controller-manager --address=127.0.0.1 --allocat>
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.287195089+02:00" level=info msg="Node token is available at /var/lib/rancher/k3s/server/token"
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.287239157+02:00" level=info msg="To join node to cluster: k3s agent -s https://192.168.100.100>
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.288561539+02:00" level=info msg="Wrote kubeconfig /etc/rancher/k3s/k3s.yaml"
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.288702330+02:00" level=info msg="Run: k3s kubectl"
Mai 01 19:14:45 k3s-master1 k3s[1065]: time="2021-05-01T19:14:45.288813900+02:00" level=fatal msg="failed to find cpu cgroup (v2)"
Mai 01 19:14:45 k3s-master1 systemd[1]: k3s.service: Main process exited, code=exited, status=1/FAILURE
░░ Subject: Unit process exited
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ An ExecStart= process belonging to unit k3s.service has exited.
░░
░░ The process' exit code is 'exited' and its exit status is 1.
Mai 01 19:14:45 k3s-master1 systemd[1]: k3s.service: Failed with result 'exit-code'.
░░ Subject: Unit failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ The unit k3s.service has entered the 'failed' state with result 'exit-code'.
Mai 01 19:14:45 k3s-master1 systemd[1]: Failed to start Lightweight Kubernetes.
░░ Subject: A start job for unit k3s.service has failed
░░ Defined-By: systemd
░░ Support: https://www.debian.org/support
░░
░░ A start job for unit k3s.service has finished with a failure.

```

This is the LXC configuration:
```
lxc.cgroup.relative = 0
lxc.cgroup.dir.monitor = lxc.monitor/111
lxc.cgroup.dir.container = lxc/111
lxc.cgroup.dir.container.inner = ns
lxc.arch = amd64
lxc.include = /usr/share/lxc/config/debian.common.conf
lxc.include = /usr/share/lxc/config/debian.userns.conf
lxc.seccomp.profile = /usr/share/lxc/config/pve-userns.seccomp
lxc.apparmor.profile = generated
lxc.apparmor.raw = deny mount -> /proc/,
lxc.apparmor.raw = deny mount -> /sys/,
lxc.mount.auto = sys:mixed
lxc.monitor.unshare = 1
lxc.idmap = u 0 100000 65536
lxc.idmap = g 0 100000 65536
lxc.tty.max = 2
lxc.environment = TERM=linux
lxc.uts.name = k3s-master2
lxc.cgroup.memory.limit_in_bytes = 2097152000
lxc.cgroup.memory.memsw.limit_in_bytes = 2107637760
lxc.cgroup.cpu.shares = 1024
lxc.rootfs.path = /var/lib/lxc/111/rootfs
lxc.net.0.type = veth
lxc.net.0.veth.pair = veth111i0
lxc.net.0.hwaddr = 9E:75:FE:E8:E7:76
lxc.net.0.name = eth0
lxc.net.0.script.up = /usr/share/lxc/lxcnetaddbr
lxc.cgroup.cpuset.cpus = 0,2
lxc.apparmor.profile: unconfined
lxc.cgroup.devices.allow: a
lxc.cap.drop: 
lxc.mount.auto: "proc:rw sys:rw"
```
The LXC is running packages of Debian Testing.

My understanding is that you already managed to deploy k3s on LXC, at least [this](https://asciinema.org/a/qeoCThX5M8dt7CFxQQjZGfTBN) video indicates this.
If you could share the requirements for this one could evaluate what is missing in Proxmox.

* What OS or type or VM are you using for your cluster? Where is it hosted? (for `k3sup install/join`):
Proxmox 6.3.6
* Operating System and version (e.g. Linux, Windows, MacOS):
Debian Buster

## "Be part of the solution"

Subject to approval, are you willing to work on a Pull Request for this issue or feature request?
No

## Are you a GitHub Sponsor (Yes/No?)
No

https://github.com/sponsors/alexellis

**Comment by cmonty14** [comment]
Fixed using Ubuntu 20.04 (and not Debian 10).

**Comment by alexellis** [comment]
I would not expect K3s to work on LXC for various reasons, but glad you got it working. Again, like I said on your other issue, k3sup is an installer, it is not K3s, for K3s, you need to speak to Rancher Labs who created and maintain K3s itself.
